avr-hal-generic/src/usart.rs

```diff
@@ -413,6 +478,14 @@ impl<H, USART: UsartOps<H, RX, TX>, RX, TX, CLOCK> UsartWriter<H, USART, RX, TX,
             _h: marker::PhantomData,
         }
     }
+
+    /// Transmit a byte.
+    ///
+    /// This method will block until the byte has been enqueued for transmission but **not** until
+    /// it was entirely sent.
+    fn write_byte(&mut self, byte: u8) {
+        nb::block!(self.p.raw_write(byte)).unwrap()
+    }
```

Comment:
Unsure whether it wouldn't be more readable to simply call ` nb::block!(self.p.raw_write(byte))` in the `write()` implementation directly to remove the additonal layer.